app/lib/features/news/pages/news_list_page.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/add_button_with_can_permission.dart';
+import 'package:acter/common/widgets/empty_state_widget.dart';
+import 'package:acter/common/widgets/space_name_widget.dart';
+import 'package:acter/features/news/providers/news_providers.dart';
+import 'package:acter/features/news/widgets/news_full_view.dart';
+import 'package:acter/features/news/widgets/news_grid_view.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::news::list');
+
+enum NewsViewMode { gridView, fullView }
+
+class NewsListPage extends ConsumerStatefulWidget {
+  final String? spaceId;
+  final NewsViewMode newsViewMode;
+
+  const NewsListPage({
+    super.key,
+    this.spaceId,
+    this.newsViewMode = NewsViewMode.gridView,
+  });
+
+  @override
+  ConsumerState<NewsListPage> createState() => _NewsListPageState();
+}
+
+class _NewsListPageState extends ConsumerState<NewsListPage> {
+  final ValueNotifier<bool> useGridMode = ValueNotifier(true);
+  final ValueNotifier<int> currentIndex = ValueNotifier(0);
+
+  @override
+  void initState() {
+    super.initState();
+    useGridMode.value = widget.newsViewMode == NewsViewMode.gridView;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: useGridMode,
+      builder: (context, value, child) {
+        return Scaffold(
+          extendBodyBehindAppBar: !value,
+          appBar: _buildAppBar(value),
+          body: _buildBody(value),
+        );
+      },
+    );
+  }
+
+  AppBar _buildAppBar(bool useGridMode) {
+    final spaceId = widget.spaceId;
+    final canPop = widget.newsViewMode == NewsViewMode.gridView &&
+        this.useGridMode.value == true;
+    return AppBar(
+      backgroundColor: Colors.transparent,
+      centerTitle: false,
+      leading: widget.newsViewMode == NewsViewMode.gridView
+          ? IconButton(
+              onPressed: () {
+                if (canPop) {
+                  Navigator.pop(context);
+                } else {
+                  this.useGridMode.value = true;
+                }
+              },
+              icon: const Icon(Icons.arrow_back),
+            )
+          : const SizedBox.shrink(),
+      title: widget.newsViewMode == NewsViewMode.gridView
+          ? Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Text(L10n.of(context).boosts),
+                if (spaceId != null) SpaceNameWidget(spaceId: spaceId),
+              ],
+            )
+          : const SizedBox.shrink(),
+      actions: [
+        AddButtonWithCanPermission(
```

Comment:
We can't do `if (spaceId != null)` as listing screens (News List, Pins List, Event List and Task List) are generic and used for for space specific list and all list.

If we add space `if (spaceId != null)` then it will not all add button at all even when user going to all list from dashboard.

If you caught very good thing. Actually `AddButtonWithCanPermission` need correction. This button is only checking `canString` with `hasSpaceWithPermissionProvider` but it should also check with membership if `spaceId` is given.

Did require modification in below commit.
https://github.com/acterglobal/a3/pull/2194/commits/573db66aff29eb53f1fecf6393c75f7cc43ad2c7